src/models/player.rs

```diff
@@ -2,14 +2,16 @@ use rand::Rng;
 
 use geometry::{Point, Size};
 use super::PowerupKind;
+use super::{Resource, Heat};
 use geometry::{Advance, Collide, Vector, Position};
 
 /// The `Player` is the rocket controlled by the user
 #[derive(Default)]
 pub struct Player {
     pub vector: Vector,
     pub is_dead: bool,
-    pub powerup: Option<PowerupKind>
+    pub powerup: Option<PowerupKind>,
+    pub resource: Resource
```

Comment:
I think this field should have a clearer name. We could call it `gun`, for instance